test/e2e/flask/multichain-api/create-session.spec.ts

```diff
@@ -46,4 +61,144 @@ describe('Multichain API', function () {
       );
     });
   });
+
+  describe('Connect wallet to the multichain dapp via `externally_connectable`, call `wallet_createSession` with requested EVM scope that match the user’s enabled networks', function () {
+    it('should only select the specified EVM scopes requested by the user, and session scope contain valid accounts only', async function () {
+      await withFixtures(
+        {
+          title: this.test?.fullTitle(),
+          fixtures: new FixtureBuilder().withPopularNetworks().build(),
+          ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+        },
+        async ({
+          driver,
+          extensionId,
+        }: {
+          driver: Driver;
+          extensionId: string;
+        }) => {
+          const INVALID_ACCOUNT = '0x9999999999999999999999999999999999999999';
+          const requestScopesToNetworkMap = {
+            'eip155:1': 'Ethereum Mainnet',
+            'eip155:42161': 'Arbitrum One',
+            'eip155:10': 'OP Mainnet',
+          };
+
+          const requestScopes = Object.keys(requestScopesToNetworkMap);
+          const networksToRequest = Object.values(requestScopesToNetworkMap);
+
+          await openMultichainDappAndConnectWalletWithExternallyConnectable(
+            driver,
+            extensionId,
+          );
+
+          await addRequestAccountsToCreateSession(driver, [
+            DEFAULT_FIXTURE_ACCOUNT,
+            INVALID_ACCOUNT,
+          ]);
+
+          await initCreateSessionScopes(driver, requestScopes);
+
+          // navigate to network selection screen
+          const editButtons = await driver.findElements('[data-testid="edit"]');
+          await editButtons[1].click();
+          await driver.delay(largeDelayMs);
+
+          await assertOnlyRequestedNetworksAreSelected(
+            driver,
+            networksToRequest,
+          );
+
+          // proceed with network selection confirm
+          await driver.clickElement(
+            '[data-testid="connect-more-chains-button"]',
```

Comment:
I know this isn't your doing but this is a very confusingly named test id if this is the confirm button...